pkg/handler/v2/alertmanager/alertmanager_test.go

```diff
@@ -0,0 +1,410 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alertmanager_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	apiv2 "k8c.io/kubermatic/v2/pkg/api/v2"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/test"
+	"k8c.io/kubermatic/v2/pkg/handler/test/hack"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	testAlertmanagerConfigSecretName = "test-alertmanager"
+	testAlertmanagerConfig           = `
+alertmanager_config: |
+  global:
+    smtp_smarthost: 'localhost:25'
+    smtp_from: 'test@example.org'
+  route:
+    receiver: "test"
+  receivers:
+    - name: "test"
+      email_configs:
+      - to: 'test@example.org'
+`
+)
+
+func TestGetEndpoint(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		Name                      string
+		ProjectID                 string
+		ClusterID                 string
+		ExistingKubermaticObjects []ctrlruntimeclient.Object
+		ExistingAlertmanager      *kubermaticv1.Alertmanager
+		ExistingConfigSecret      *corev1.Secret
+		ExistingAPIUser           *apiv1.User
+		ExpectedResponse          apiv2.Alertmanager
+		ExpectedHTTPStatus        int
+	}{
+		{
+			Name:      "scenario 1: get alertmanager that belongs to the given cluster",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			ExistingAlertmanager: test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
+				testAlertmanagerConfigSecretName),
```

Comment:
NIT: you can just put this up as its a KubermaticObject (disregard the weird formatting for some reason i couldnt make it nice)
```suggestion
			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
				test.GenTestSeed(),
				test.GenDefaultCluster(),
				test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
	                        testAlertmanagerConfigSecretName),
			),
```

Reply:
You are right, fixed